protected/humhub/modules/content/widgets/richtext/converter/BaseRichTextConverter.php

```diff
@@ -169,7 +169,10 @@ public static function process($text, $options = [])
      */
     public static function buildCacheKeyForContent(ContentOwner $content, $prefix = null)
     {
-        $result = 'content_'.$content->content->id;
+        $result = 'content_' . $content->content->id;
+        if (isset($content->id)) {
+            $result .= '_' . $content->id;
```

Comment:
We always need the id of the content and not the record here. Otherwise a wiki page with id 1 will return the same result as post with id 1. The `ContentOwner` interface should always return a `Content` by `$content->content` unfortunately the interface is broken with content active records since it returns an activequery instead of the content itself when calling `getContent()`

Reply:
Note there is also a `buildCacheKeyForRecord` which will include the record class.

Reply:
@yurabakhtin I think the following would solve the issue:

```
if($content instanceof ContentAddonActiveRecord) {
    return static::buildCacheKeyForRecord($content);
}

// revert the rest back to prior this PR
```

Reply:
@buddh4 Yes, your solution works well, so I applied this here https://github.com/humhub/humhub/pull/4882/commits/7c31388d80703f3643b4e11fc81fd32862162532 and also reverted there the fix from PR https://github.com/humhub/humhub/pull/4870/files